service/tbc/rpc.go

```diff
@@ -495,6 +509,49 @@ func (s *Server) handleTxByIdRequest(ctx context.Context, req *tbcapi.TxByIdRequ
 	}, nil
 }
 
+func (s *Server) handleTxBroadcastRequest(ctx context.Context, req *tbcapi.TxBroadcastRequest) (any, error) {
+	log.Tracef("handleTxBroadcastRequest")
+	defer log.Tracef("handleTxBroadcastRequest exit")
+
+	txid, err := s.TxBroadcast(ctx, req.Tx, req.Force)
+	if err != nil {
+		var responseErr *protocol.Error
+		if errors.Is(err, ErrTxAlreadyBroadcast) || errors.Is(err, ErrTxBroadcastNoPeers) {
+			responseErr = protocol.RequestError(err)
+		} else {
+			responseErr = protocol.NewInternalError(err).ProtocolError()
+		}
+		return &tbcapi.TxBroadcastResponse{Error: responseErr}, responseErr
+	}
+
+	return &tbcapi.TxBroadcastResponse{TxID: txid}, nil
+}
+
+func (s *Server) handleTxBroadcastRawRequest(ctx context.Context, req *tbcapi.TxBroadcastRawRequest) (any, error) {
+	log.Tracef("handleTxBroadcastRawRequest")
+	defer log.Tracef("handleTxBroadcastRawRequest exit")
+
+	tx := wire.NewMsgTx(0)
+	err := tx.Deserialize(bytes.NewBuffer(req.Tx))
+	if err != nil {
+		return &tbcapi.TxBroadcastResponse{
+			Error: protocol.RequestError(err),
+		}, nil
+	}
+	txid, err := s.TxBroadcast(ctx, tx, req.Force)
+	if err != nil {
+		var responseErr *protocol.Error
+		if errors.Is(err, ErrTxAlreadyBroadcast) || errors.Is(err, ErrTxBroadcastNoPeers) {
+			responseErr = protocol.RequestError(err)
+		} else {
+			responseErr = protocol.NewInternalError(err).ProtocolError()
+		}
+		return &tbcapi.TxBroadcastResponse{Error: responseErr}, responseErr
```

Comment:
The returned error should be the error itself, not a protocol error.
 `ProtocolError()` on an internal error will also only contain "internal error", and won't include the traces or anything useful for logging.
```suggestion
		if errors.Is(err, ErrTxAlreadyBroadcast) || errors.Is(err, ErrTxBroadcastNoPeers) {
			return &tbcapi.TxBroadcastResponse{Error: protocol.RequestError(err)}, err
		}
		e := protocol.NewInternalError(err)
		return &tbcapi.TxBroadcastResponse{Error: e.ProtocolError()}, e
```